dataGenerator.js

```diff
@@ -88,182 +52,36 @@ while (numberOfReviews < numberToWrite) {
-    } else if (numberOfReviews / numberToWrite < .8) {
-      reviews8 += `${numberOfReviews + 1},${product_id + 1},${random(1, 6)},${date.toString().slice(0, 24)},${faker.random.words()},${faker.lorem.paragraph()},${(Math.random() >= .3)},${false},${faker.name.findName()},${faker.internet.email()},${"none"},${random(0, 30)}\n`
-    } else if (numberOfReviews / numberToWrite < .9) {
-      reviews9 += `${numberOfReviews + 1},${product_id + 1},${random(1, 6)},${date.toString().slice(0, 24)},${faker.random.words()},${faker.lorem.paragraph()},${(Math.random() >= .3)},${false},${faker.name.findName()},${faker.internet.email()},${"none"},${random(0, 30)}\n`
-    } else if (numberOfReviews / numberToWrite <= 1) {
-      reviews10 += `${numberOfReviews + 1},${product_id + 1},${random(1, 6)},${date.toString().slice(0, 24)},${faker.random.words()},${faker.lorem.paragraph()},${(Math.random() >= .3)},${false},${faker.name.findName()},${faker.internet.email()},${"none"},${random(0, 30)}\n`
-    }
+    reviews += `${numberOfReviews + 1},${product_id + 1},${random(1, 6)},${date.toString().slice(0, 24)},${faker.random.words()},${faker.lorem.paragraph()},${(Math.random() >= .3)},${false},${faker.name.findName()},${faker.internet.email()},${"none"},${random(0, 30)}\n`
 
     // generate values for this review's characteristics + add to csv
     for (var j = 0; j < products[product_id].characteristics.length; j++) {
       let characteristic = `${characteristicsCounter},${products[product_id].characteristics[j]},${numberOfReviews + 1},${random(1, 6)}\n`
-
-      if (numberOfReviews / numberToWrite < .1) {
-        characteristics1 += characteristic;
-      } else if (numberOfReviews / numberToWrite < .2) {
-        characteristics2 += characteristic;
-      } else if (numberOfReviews / numberToWrite < .3) {
-        characteristics3 += characteristic;
-      } else if (numberOfReviews / numberToWrite < .4) {
-        characteristics4 += characteristic;
-      } else if (numberOfReviews / numberToWrite < .5) {
-        characteristics5 += characteristic;
-      } else if (numberOfReviews / numberToWrite < .6) {
-        characteristics6 += characteristic;
-      } else if (numberOfReviews / numberToWrite < .7) {
-        characteristics7 += characteristic;
-      } else if (numberOfReviews / numberToWrite < .8) {
-        characteristics8 += characteristic;
-      } else if (numberOfReviews / numberToWrite < .9) {
-        characteristics9 += characteristic;
-      } else if (numberOfReviews / numberToWrite <= 1) {
-        characteristics10 += characteristic;
-      }
-
+      characteristics += characteristic;
       characteristicsCounter++;
     }
 
     // generate photos for this review + add to csv
     let photoNumber = random(0, 8);
     for (var a = 0; a < photoNumber; a++) {
       let reviewPhoto = `${photoCounter},${numberOfReviews + 1},${faker.image.imageUrl()}\n`;
```

Comment:
why do you need photoCounter? is it to provide a unique ID? If so, did Josh say that it would be faster to insert a unique ID on the DB side? I am not sure on this but I wonder if thing would be more efficient if you omit the counters here and put them in on the database side...